File: infra/src/main/java/com/beat/infra/persistence/cast/mapper/CastPersistenceMapper.java
package com.beat.infra.persistence.cast.mapper;

import org.springframework.stereotype.Component;

import com.beat.domain.performance.model.Cast;
import com.beat.infra.persistence.cast.entity.CastJpaEntity;

@Component
public class CastPersistenceMapper {

	public Cast toDomain(CastJpaEntity entity) {
		return Cast.rehydrate(
			entity.getId(),
			entity.getCastName(),
			entity.getCastRole(),
			entity.getCastPhoto()
		);
	}

	public CastJpaEntity toEntity(Cast cast, long performanceId) {
		return CastJpaEntity.rehydrate(
			cast.getId(),
			cast.getCastName(),
			cast.getCastRole(),
			cast.getCastPhoto(),
			performanceId
		);
	}
}

File: infra/src/main/kotlin/com/beat/infra/persistence/cast/mapper/CastPersistenceMapper.kt
package com.beat.infra.persistence.cast.mapper

import com.beat.domain.performance.model.Cast
import com.beat.infra.persistence.cast.entity.CastJpaEntity
import org.springframework.stereotype.Component

@Component
class CastPersistenceMapper {
    fun toDomain(entity: CastJpaEntity): Cast = Cast.rehydrate(
        entity.id,
        entity.castName,
        entity.castRole,
        entity.castPhoto,
    )

    fun toEntity(cast: Cast, performanceId: Long): CastJpaEntity = CastJpaEntity.rehydrate(
        cast.getId(),
        cast.castName,
        cast.castRole,
        cast.castPhoto,
        performanceId,
    )
}
